testsuite/tests/kuadrant/authorino/metrics/conftest.py

```diff
@@ -0,0 +1,55 @@
+"""Conftest for the Authorino metrics tests"""
+import time
+from importlib import resources
+
+import pytest
+
+from testsuite.openshift.objects.metrics import Prometheus
+
+
+@pytest.fixture(scope="session")
+def prometheus(openshift):
+    """
+    Return an instance of OpenShift metrics client
+    Skip tests if query route is not properly configured
+    """
+    # find thanos-querier route in the openshift-monitoring project
+    # this route allows to query metrics
+    openshift_monitoring = openshift.change_project("openshift-monitoring")
+    routes = openshift_monitoring.routes.for_service("thanos-querier")
+    if len(routes) > 0:
+        url = ("https://" if "tls" in routes[0]["spec"] else "http://") + routes[0]["spec"]["host"]
+        return Prometheus(url, openshift.token, openshift.project)
+
+    return pytest.skip("Skipping metrics tests as query route is not properly configured")
+
+
+@pytest.fixture(scope="module")
+def label_metrics_service(authorino, module_label):
+    """Label Authorino controller-metrics service for the proper discovery"""
+    authorino.metrics_service.label({"app": module_label})
+
+
+# pylint: disable=unused-argument
+@pytest.fixture(scope="module", autouse=True)
+def create_service_monitor(request, openshift, authorino, blame, module_label, prometheus, label_metrics_service):
+    """Create ServiceMonitor object to follow Authorino /metrics and /server-metrics endpoints"""
+    with resources.path("testsuite.resources", "service-monitor.yaml") as path:
```

Comment:
While this is not neccessarily wrong. You could create an OpenshiftObject descendant for ServiceMonitor instead

Reply:
The new Service Monitor object resides under `testsuite/openshift/objects/metrics/service_monitor.py`